### PR TITLE
fix(spanner): Explicit declaration of dependencies for OpenTelemetry traces sample

### PR DESCRIPTION
**Pull request by nareshz**
This PR includes changes to add all the required dependencies for OpenTelemetry traces sample explicitly.

### CODE REVIEW

**Comment by Sita04** [comment]
@minherz PTAL at these changes.

**Comment by nareshz** [comment]
@olavloite Can you please take a look at your convenience?

File: spanner/opentelemetry_traces/pom.xml [hunk 1]
```diff
@@ -68,10 +76,18 @@
       <groupId>com.google.cloud</groupId>
       <artifactId>google-cloud-spanner</artifactId>
     </dependency>
+    <dependency>
+      <groupId>io.opentelemetry</groupId>
+      <artifactId>opentelemetry-api</artifactId>
+    </dependency>
```
Comment (minherz) [inline spanner/opentelemetry_traces/pom.xml:82]: why this dependency is included two times (see lines 52-55)?

**Review by minherz — COMMENTED** [review]
what is a reason for adding all these dependencies?
is it the best practice to be followed?

File: spanner/opentelemetry_traces/pom.xml (hunk 1 above)
Comment (nareshz) [reply spanner/opentelemetry_traces/pom.xml:82]: In documentation, we want to show dependencies required for using Cloud Trace exporter and OTLP exporter seperately. That's why two tags are defined in this pom.xml file - spanner_opentelemetry_traces_otlp_dependencies(otlp exporter dependencies) and spanner_opentelemetry_traces_cloudtrace_dependencies(cloud trace exporter dependencies). There maybe common dependency across both.

**Comment by nareshz** [comment]
> what is a reason for adding all these dependencies? is it the best practice to be followed?

Yes, right. The reason is mainly to follow good practice of explicitly declaring all the dependencies required.

**Comment by minherz** [comment]
> > what is a reason for adding all these dependencies? is it the best practice to be followed?
> 
> Yes, right. The reason is mainly to follow good practice of explicitly declaring all the dependencies required.

I am not aware of such good practice. Can you please reference me to Google documentation that defines it? If not, please provide another widely accepted guidance reference.

Thank you

File: spanner/opentelemetry_traces/pom.xml [hunk 2]
```diff
@@ -49,10 +49,18 @@
       <groupId>com.google.cloud</groupId>
       <artifactId>google-cloud-spanner</artifactId>
     </dependency>
+    <dependency>
+      <groupId>io.opentelemetry</groupId>
+      <artifactId>opentelemetry-api</artifactId>
```
Comment (surbhigarg92) [inline spanner/opentelemetry_traces/pom.xml:54]: This dependency is already part of Spanner Java Client library. Why do we need to add this again here ?

File: spanner/opentelemetry_traces/pom.xml [hunk 3]
```diff
@@ -49,10 +49,18 @@
...
+    </dependency>
     <dependency>
       <groupId>io.opentelemetry</groupId>
       <artifactId>opentelemetry-sdk</artifactId>
     </dependency>
+    <dependency>
+      <groupId>io.opentelemetry</groupId>
+      <artifactId>opentelemetry-sdk-common</artifactId>
```
Comment (surbhigarg92) [inline spanner/opentelemetry_traces/pom.xml:62]: Can you please help me remind and what was this needed for ?

Comment (nareshz) [reply spanner/opentelemetry_traces/pom.xml:62]: This is needed for the `io.opentelemetry.sdk.resources.Resource` class.

File: spanner/opentelemetry_traces/pom.xml (hunk 2 above)
Comment (nareshz) [reply spanner/opentelemetry_traces/pom.xml:54]: As discussed offline with you @surbhigarg92 that we should only declare dependencies that needs to be added by customers, not making any changes to explicitly declare the dependencies.

File: spanner/opentelemetry_traces/pom.xml (hunk 1 above)
Comment (minherz) [reply spanner/opentelemetry_traces/pom.xml:82]: The marked dependencies include the artifacts with the same groupId and artifactId. It does not matter how many components use it. They all have to use **the same** dependency. Creating multiple dependencies of the same artifact can lead to using several versions of the artifact. It is bad practice and should not be used.

**Pull request closed without merging**